sympy/solvers/solvers.py

```diff
@@ -350,7 +350,11 @@ def check_assumptions(expr, against=None, **assumptions):
         if test is expected:
             continue
         elif test is not None:
-            return False
+            if get is True:
+                return False, key
```

Comment:
Yes I looked in the particular file, and found the following code:

```
def test_check_assumptions():
    x = symbols('x', positive=True)
    assert solve(x**2 - 1) == [1]
    assert check_assumptions(1, x) == True
```

In this I am not able to understand `assert solve(x**2 - 1) == [1]`  and `assert check_assumptions(1, x) == True` . How they are able to test the cases?